app/controllers/admin/sites_controller.rb

```diff
@@ -76,7 +76,7 @@ def destroy
     private
 
     def set_site
-      @site = Site.friendly.find(params[:id])
+      @site ||= Site.friendly.find(params[:id])
```

Comment:
How does this change relate to the rest of this PR? Doesn't seem to be part of the changes here.

Reply:
The sites controller before_action logic is a little tricky. This line is from a method that is invoked before update and destroy action. It set the site (duh) but it also *reset* the site if it was called more than once.

In the update action if the save fails it renders the edit view with the feedback. The view has some values that need to be set up in `set_variables_for_sites_neighbourhoods_selection` (brilliant name) which in turn calls `set_site` and therefore any errors on the `@site` object will be lost.

The PR is about missing error feedback on admin pages and this line was responsible for resetting the site (as set by the user provided params).

Reply:
Ah. The commit message and PR description implied this was just about writing tests, so seeing non-test changes confused me. Thanks for explaining.

Reply:
I see your point. It was writing tests but when writing those tests I found bugs that needed to be fixed. literal tdd.

I will write a better merge description